libs/networking/http_helpers/http_helpers.ml

```diff
@@ -0,0 +1,122 @@
+open Cohttp
+
+(*****************************************************************************)
+(* Prelude *)
+(*****************************************************************************)
+(* A few helpers to perform http GET and POST requests.
+ *
+ * Below we separate the methods out by async (returns Lwt promise),
+ * and sync (runs async method in lwt runtime)
+ * This way we can use the async methods in the language server,
+ * and other places too.
+ *
+ * Note that using [@@profiling] with xxx_async function is useless
+ * as the actual computation is done in the caller doing the
+ * Lwy_main.run
+ *)
+
+(*****************************************************************************)
+(* Client *)
+(*****************************************************************************)
+
+(* Create a client reference so we can swap it out with a testing version *)
+(*****************************************************************************)
+(* Async *)
+(*****************************************************************************)
+let client_ref : (module Cohttp_lwt.S.Client) option ref = ref None
+
+module Make (Lwt_wrapper : sig
```

Comment:
Why we don't use the same technique than above and use a ref to a first class module like lwt_wrapper_ref ?
I guess we can't use a ref to the run function, because it's polymorphic and OCaml usually complains about
ref to polymorphic stuff, but a ref to a first class module maybe would work.

I just hate functors.

Reply:
Will add a comment though